### PR TITLE
Add --use-system-phantomjs flag

### PR DESCRIPTION
**Pull request by benweissmann**
I'd like to be able to use spacejam, but I need to use phantomjs 2.0. This PR adds support for a `--use-system-phantomjs` flag that causes spacejam to not modify `process.env.PATH` before running `phantomjs`, so users that want to use their own build of phantomjs can do so.

### CODE REVIEW

File: src/Phantomjs.coffee [hunk 1]
```diff
@@ -4,15 +4,16 @@ ChildProcess = require './ChildProcess'
 EventEmitter = require('events').EventEmitter
 path = require 'path'
 phantomjs = require 'phantomjs'
-process.env.PATH = path.dirname(phantomjs.path) + ':' + process.env.PATH
 isCoffee = __filename.indexOf('.coffee') > 0
 
+DEFAULT_PATH = process.env.PATH
```
Comment (rbabayoff) [inline src/Phantomjs.coffee]: Don't set it here, just set process.env.PATH inside Phantomjs, so there are no test side effects

File: tests/src/CLITest.coffee [hunk 2]
```diff
@@ -84,3 +90,32 @@ describe "CLI", ->
...
+    testPackagesStub.restore()
+    process.chdir(__dirname + "/../apps/leaderboard/packages/success")
+    # We set mongo-url to mongodb:// so test will be faster
+    process.argv.push 'test-packages', '--mongo-url', 'mongodb://', '--use-system-phantomjs', '--phantomjs-options=--ignore-ssl-errors=true --load-images=false', './'
+    cli.exec()
+    spacejam.on 'done', (code)=>
+      try
+        if code is 0 then done() else done("spacejam.done=#{code}")
+
+        expect(process.env.PATH).to.equal(DEFAULT_PATH)
```
Comment (rbabayoff) [inline tests/src/CLITest.coffee]: Save process.env.PATH before call to cli.exec and compare after.

File: src/Phantomjs.coffee [hunk 3]
```diff
@@ -34,6 +36,12 @@ class Phantomjs extends EventEmitter
       env: env
     log.debug 'spawnOptions:', spawnOptions
 
+    # Add phantomjs NPM package bin to PATH unless --use-system-phantomjs is passed
+    if useSystemPhantomjs
```
Comment (rbabayoff) [inline src/Phantomjs.coffee]: This should change to:

```
if not ...
  ...
```
And nothing else.

**Comment by rbabayoff** [comment]
Disregard my comments. All good. Can you please run cake test from the command line to see all tests pass for pure coffee too? In travis, I compile to js and only run the tests in pure node. Reason I didn't upgrade to phantomjs 2.0 is because native coffee support is discontinued, and I'm using the phantomjs coffee script when running cake coffee.

**Comment by benweissmann** [comment]
Yes, both `cake test` and `npm test` succeed from the command line.

**Comment by rbabayoff** [comment]
Published as v1.2.1

**Pull request merged**